### PR TITLE
CIV-10893 - Court Calculator Trigger and Navigation

### PR DESCRIPTION
**Pull request by neeta-hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-10893

### Change description ###
Integrated the courtDecision API and based upon calculated decision user will navigate to the respective screens.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by jeswanth-hmcts — DISMISSED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

File: src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const fields ={
 };
 
 const newDate = new Date(new Date().setMonth(new Date().getMonth()+2));
```
Comment (jarekPierchala) [inline src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js:18]: +2 twice?

Comment (jarekPierchala) [reply src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js:18]: const newDate =new Date(new Date().setMonth(new Date().getMonth()+2));
and 
const month = newDate.getMonth()+2;

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

File: src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js (hunk 1 above)
Comment (neeta-hmcts) [reply src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js:18]: just a temporary changes for FT fix. Later Raja will add a proper fix.

Comment (jeswanth-hmcts) [reply src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js:18]: can't we use this method  existing method for adding months src\main\common\utils\dateUtils.ts   addMonths

Comment (neeta-hmcts) [reply src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js:18]: > can't we use existing method for adding months src\main\common\utils\dateUtils.ts addMonths

ok sure.. I will ask raja to check this while implementing proper fix.

Comment (neeta-hmcts) [reply src/test/functionalTests/features/response/pages/defendantLipResponse/defendantDQ/availabilityDates.js]: Thanks Jeswanth for adding the fix.

**Review by jeswanth-hmcts — DISMISSED** [review]

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts [hunk 2]
```diff
@@ -134,8 +134,8 @@ export function getProposeAlternativeRepaymentTask(claim: Claim, claimId: string
     url: constructResponseUrlWithIdParams(claimId, CLAIMANT_RESPONSE_PAYMENT_OPTION_URL),
     status: TaskStatus.INCOMPLETE,
   };
-  if ((claim.isPAPaymentOptionPayImmediately() && claim.claimantResponse?.courtProposedDate?.decision) ||
-    (claim.isPAPaymentOptionByDate() && claim.partialAdmission?.paymentIntention?.paymentDate)) {
+  if ((claim.respondent1.type === 'ORGANISATION' || claim.respondent1.type === 'COMPANY')
```
Comment (jarekPierchala) [inline src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts:137]: `claim.respondent1.type === 'ORGANISATION' || claim.respondent1.type === 'COMPANY')
     || claim.claimantResponse?.suggestedPaymentIntention?.paymentOption !== undefined`
     
     To separate method please

**Review by jarekPierchala — CHANGES_REQUESTED** [review]

**Review by laarmada — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

File: src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts (hunk 2 above)
Comment (laarmada) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts:137]: you can use existing function:
image
On the other hand that is not necessary: 
image

Comment (neeta-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseTasklistService/claimantResponseTasks/whatToDoNextSectionTasks.ts:137]: Yes.. I will fix this in my ongoing ticket CIV-8785 . Considering this PR will unblock others ticket.

**Pull request merged**